lib/mongo/crypt/kms/azure.rb

```diff
@@ -105,7 +105,7 @@ class MasterKeyDocument
         #
         # @raise [ ArgumentError ] If required options are missing or incorrectly.
           def initialize(opts)
-            if opts.is_a?(Hash)
+            unless opts.is_a?(Hash)
```

Comment:
Interesting...

Reply:
Yeah, in the previous PR there were no test for Data Key for Azure. Therefore this error got through. In this PR I added missing tests for Azure, and for GCP.

Reply:
Fixed